gcloud/pubsub/topic.py

```diff
@@ -113,6 +122,8 @@ def publish(self, message, **attrs):
         :rtype: str
         :returns: message ID assigned by the server to the published message
         """
+        if self.add_timestamp_to_messages:
+            attrs['timestamp'] = '%sZ' % _NOW().isoformat()
```

Comment:
I'm torn on this.

The hypothetical user goes from being confused that their value was overwritten to confused that their setting `timestamp_messages=True` is not working as expected.

By throwing we say "hey, we don't support doing both of these. you need to pick one"